app/pages/sign-up.tsx

```diff
@@ -17,38 +54,94 @@ const SignUp = () => {
     setBenefitRefs((prev) => [...prev, { current: null }]);
   }, []);
 
-  const handleCreateAccount = useCallback(() => {
+  // Only call if wallet connected.
+  const handleCreateAccount = useCallback(async () => {
     const username = usernameRef?.current?.value;
     const email = emailRef?.current?.value;
     const description = descriptionRef?.current?.value;
 
-    const benefits = benefitRefs.map((benefitRef) => {
+    let benefits = benefitRefs.map((benefitRef) => {
       return benefitRef?.current?.value;
     });
 
+    // Filter out blank benefits
+    benefits = benefits.filter((benefit) => benefit);
```

Comment:
you can chain the `.map` and `.filter` (i.e. `.map(---).filter(---)`)so you don't have to do the `let` and reassigning variable thing